pkg/yang/entry.go

```diff
@@ -559,11 +559,18 @@ func ToEntry(n Node) (e *Entry) {
 		}
 	}
 	ms := RootNode(n).Modules
-	if e := ms.entryCache[n]; e != nil {
+
+	// Protect the entryCache map from concurrent access.
+	ms.entryCacheMu.RLock()
+	e = ms.entryCache[n]
+	ms.entryCacheMu.RUnlock()
+	if e != nil {
 		return e
 	}
 	defer func() {
+		ms.entryCacheMu.Lock()
 		ms.entryCache[n] = e
+		ms.entryCacheMu.Unlock()
```

Comment:
`entryCache` is also modified/read at the following location:

```
326         ms.entryCache = map[Node]*Entry{}
```

I think it's cleaner if this is done:
* Define functions SetEntryCache() and GetEntryCache() and use them here.
* At line 326, call `ClearEntryCache`
* Add a comment to the field `entryCache` that it SHOULD be accessed using the getter/setter/clearer, rather than using the locks directly.

Reply:
Thanks @wenovus for the review. Yes, that's much cleaner - I added the `getEntryCache()`/`setEntryCache()` functions and comment.

Reply:
Thanks!